storage/integration_test.go

```diff
@@ -636,21 +634,15 @@ func TestIntegration_BucketPolicyOnly(t *testing.T) {
 	}
 
 	// Check that the object ACLs are the same.
-	var acls []ACLRule
-	err = retry(ctx, func() error {
-		acls, err = o.ACL().List(ctx)
-		if err != nil {
-			return fmt.Errorf("ACL.List: object ACL list failed: %v", err)
-		}
-		return nil
-	}, func() error {
-		if !containsACL(acls, aclEntity, RoleReader) {
-			return fmt.Errorf("containsACL: expected ACLs %v to include custom ACL entity %v", acls, aclEntity)
-		}
-		return nil
-	})
+	ctxWithTimeout, cancelCtx = context.WithTimeout(ctx, time.Second*10)
+	acls, err := o.Retryer(WithPolicy(RetryAlways)).ACL().List(ctxWithTimeout)
+	cancelCtx()
 	if err != nil {
-		t.Fatal(err)
+		t.Errorf("ACL.List: object ACL list failed: %v", err)
+	}
+
+	if !containsACL(acls, aclEntity, RoleReader) {
```

Comment:
Unfortunately this check needs to be retried as well, since it's possible/likely that the call will succeed but the check will fail.

It might be better to leave the original code for stuff with this pattern.

Reply:
Oh interesting... so it's possible for a situation where the metadata is updated for bucket level access policy but the acls are not yet populated?

Reply:
Actually I looked at this again and I think I was wrong, I believe that as long as ACL.List() succeeds then it should contain the correct items in this instance. Feel free to revert to what you had originally!